template.go

```diff
@@ -156,11 +161,18 @@ func BuildTemplate(dir string, files ...string) error {
 		fmt.Printf("filepath.Walk() returned %v\n", err)
 		return err
 	}
+	buildAllFiles := len(files) == 0
```

Comment:
It's not much of a problem.len() is fast enough as a  builtin function and this BuildTemplate function  will be called by the init method in product environment.